common/web/types/src/keyboard-object.ts

```diff
@@ -0,0 +1,180 @@
+/*
+ * Keyman is copyright (C) SIL Global. MIT License.
+ */
+import { TouchLayoutPlatform as LayoutFormFactorSpec } from './keyman-touch-layout/keyman-touch-layout-file.js';
+
+export type ComplexKeyboardStore = (string | { t: 'd', d: number } | { ['t']: 'b' })[];
+
+// A stub for KeyEvent which is properly defined in KeymanWeb
+type KeyEventStub = {};
+
+// A stub for OutputTarget which is properly defined in KeymanWeb
+type OutputTargetStub = {};
```

Comment:
Excellent, good plan to use `Stub` here and avoid touching too much within KMW source itself.